blocks/bricks/recurrent.py

```diff
@@ -180,12 +182,21 @@ def recurrent_apply(brick, application, application_call,
             for name, state in states_given.items():
                 states_given[name] = tensor.unbroadcast(state,
                                                         *range(state.ndim))
+            if config.scan_strict:
+                # Find all shared variables
+                shared_vars = list(Selector(brick).get_params().values())
+                # Just make `scan_function` discard them before calling
+                # `application_function`
+                strict = {'strict': True}
```

Comment:
I do not see `strict` as a new keyword argument. Do you plan to add it?

Also there is something wrong with your branch that does not let the tests run, can you check why it can not be merged into `master`?